src/offer/offerService.js

```diff
@@ -26,6 +28,12 @@ export class OfferService {
     if (isCreated !== true) {
       throw new FailedCreatingError(FAILED_CREATE, OFFERS);
     }
+    try {
+      await this.inventoryService.removeItemsFromUser(userName, offer);
+    } catch (error) {
+      this.deleteOffer(id);
```

Comment:
Its correct call service deleteOffer here? because deleteOffer in service add the items to user and here the items never was remove